godot-core/src/task/futures.rs

```diff
@@ -308,8 +334,188 @@ impl<C: WithBaseField, R: ParamTuple + Sync + Send> IntoFuture for &TypedSignal<
     }
 }
 
+/// Convert a value into a type that is [`Send`] at compile-time while the value might not be.
+///
+/// This allows to turn any implementor into a type that is `Send`, but requires to also implement [`AssertSafeSend`] as well.
+/// The later trait will verify if a value can actually be sent between threads at runtime.
+#[expect(private_bounds)]
+pub trait IntoMaybeSend: Sealed {
+    type Target: Send + AssertSafeSend<Target = Self>;
+
+    fn into_maybe_send(self) -> Self::Target;
+}
+
+/// Assert that it was safe to send the value to the current thread.
+///
+/// # Safety
+/// The implementor has to guarantee that `assert_safe_send` panics if the value has been sent between threads while being `!Send`.
+#[expect(private_bounds)]
+pub unsafe trait AssertSafeSend: Send + Sealed {
+    type Target;
+
+    fn assert_safe_send(self) -> Self::Target;
+}
+
+trait Sealed {}
```

Comment:
Can you reuse this?
https://github.com/godot-rust/gdext/blob/7f5191e577c40dc830cc8dac61761a45fa92f485/godot-core/src/meta/sealed.rs#L16